pkg/kubelet/dockertools/docker.go

```diff
@@ -166,24 +166,19 @@ func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 		// No Tag or SHA specified, so just return what we have
 		return true
 	}
+
 	if isTagged {
-		hostname, _ := dockerref.SplitHostname(named)
-		// Check the RepoTags for an exact match
+		// Check the RepoTags for a match.
 		for _, tag := range inspected.RepoTags {
-			// Deal with image with hostname specified
-			if len(hostname) > 0 {
-				if strings.HasSuffix(image, tag) {
-					return true
-				}
-
-			} else {
-				if tag == image {
-					// We found a specific tag that we were looking for
-					return true
-				}
+			// An image name (without the tag/digest) can be [hostname '/'] component ['/' component]*
+			// Because either the RepoTag or the name *may* contain the
+			// hostname or not, we only check for the suffix match.
+			if strings.HasSuffix(image, tag) || strings.HasSuffix(tag, image) {
```

Comment:
Yes, they will. Docker allows users to specify an image string without the registry name (i.e., the hostname in the comment), and supplies a default registry (`docker.io`) in this case. The same applies to tagging the image. Therefore, both the image and the tag could contain the registry or not, and that's why I use the suffix check.

One way to make this more resilient is probably hard-code the default docker registry in kubelet for comparison, I entertained the idea but didn't like it too much. If you think that makes more sense, I can change this PR to do that. By the way, this is a sanity check after docker supposedly pulled the image. If the image doesn't exist, kubelet wouldn't be able to create the container whether the check passes or not.